### PR TITLE
chore: Refactor default branch codes

### LINKED ISSUE #26875: Migrate existing applications to use the new datastructure
As part of the new schema to mark a branch as default, we need to migrate the old applications to the new format. Here are the actions that we need to do for this task:

1. Create migration to migrate old applications in the new format
2. Change the create git connect flow so that `isDefaultBranch`  is set to true when application created from default branch.
3. Update the get branches API so that it uses the new property.
4. Update the code so that default application is displayed in the home page.
5. Update the delete branch code so that default branch can not be deleted
6. Open the default branch when app is opened without branch name parameter
7. Replace the `getApplicationById` calls inside Git service by `getApplicationInDefaultBranchByDefaultApplicationId`

### PR DESCRIPTION
**Pull request by nayan-rafiq**
## Description
This PR refactors the code related to list of branches and default branch. It separates different parts of the code into separate functions.

#### PR fixes following issue(s)
Fixes #26875

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

#### How Has This Been Tested?
- [x] Manual
- [x] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Review by AnaghHegde — APPROVED** [review]

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java [hunk 1]
```diff
@@ -1594,113 +1602,130 @@ public Mono<GitPullDTO> pullApplication(String defaultApplicationId, String bran
         return Mono.create(sink -> pullMono.subscribe(sink::success, sink::error, null, sink.currentContext()));
     }
 
+    private Flux<Application> updateDefaultBranchName(
+            Path repoPath, String defaultBranchName, String defaultApplicationId) {
+        // Get the application from DB by new defaultBranch name
+        return addFileLock(defaultApplicationId)
```
Comment (AnaghHegde) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java:1608]: This should not be present here. This should be moved to the `syncDefaultBranchNameFromRemote` method.

Comment (nayan-rafiq) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java]: Fixed

**Review by AnaghHegde — APPROVED** [review]

**Comment by Parthvi12** [comment]
@nayan-rafiq When I try to disconnect git from a particular app, it takes lot of time and then request times out giving this error: 
Screenshot 2023-09-13 at 2 14 50 PM

**Review by Parthvi12 — APPROVED** [review]

**Pull request merged**